CMakeLists.txt

```diff
@@ -1,30 +1,83 @@
-cmake_minimum_required(VERSION 3.1)
+cmake_minimum_required(VERSION 3.5)
 
 set(PROJECT_ORG stack-of-tasks)
 set(PROJECT_NAME dynamic_graph_bridge_msgs)
 set(PROJECT_DESCRIPTION "Dynamic graph bridge msgs ")
 set(PROJECT_URL "https://github.com/${PROJECT_ORG}/${PROJECT_NAME}")
 
 set(CUSTOM_HEADER_DIR dynamic_graph_bridge_msgs)
-set(PROJECT_USE_CMAKE_EXPORT TRUE)
+if(DEFINED ENV{ROS_VERSION} )
+  # ROS-1
+  if ($ENV{ROS_VERSION} EQUAL 1)
+    set(PROJECT_USE_CMAKE_EXPORT TRUE)
+  endif()
+endif()
 
 include(cmake/base.cmake)
 
-COMPUTE_PROJECT_ARGS(PROJECT_ARGS LANGUAGES CXX)
+COMPUTE_PROJECT_ARGS(PROJECT_ARGS LANGUAGES CXX C)
 project(${PROJECT_NAME} ${PROJECT_ARGS})
 
-# Declare build dependencies
-find_package(catkin REQUIRED COMPONENTS message_generation std_msgs)
+if(DEFINED ENV{ROS_VERSION} )
 
-# Add message files
-add_message_files(DIRECTORY msg FILES Matrix.msg Vector.msg)
+  # ROS-1
+  if ($ENV{ROS_VERSION} EQUAL 1)
+    # Declare build dependencies
+    find_package(catkin REQUIRED COMPONENTS message_generation std_msgs)
 
-# Add service files
-add_service_files(DIRECTORY srv FILES RunCommand.srv RunPythonFile.srv)
+    # Add message files
+    add_message_files(DIRECTORY msg FILES Matrix.msg Vector.msg)
 
-# And now generate the messages
-generate_messages(DEPENDENCIES ${MESSAGE_DEPENDENCIES})
+    # Add service files
+    add_service_files(
+        DIRECTORY srv
+        FILES RunCommand.srv RunPythonCommand.srv RunPythonFile.srv
+    )
 
-add_library(${PROJECT_NAME} INTERFACE)
-install(TARGETS ${PROJECT_NAME} EXPORT ${TARGETS_EXPORT_NAME} DESTINATION lib)
-install(FILES package.xml DESTINATION share/${PROJECT_NAME})
+    # And now generate the messages
+    generate_messages(DEPENDENCIES ${MESSAGE_DEPENDENCIES})
+
+    add_library(${PROJECT_NAME} INTERFACE)
+
+    install(TARGETS ${PROJECT_NAME} EXPORT ${TARGETS_EXPORT_NAME} DESTINATION lib)
+    install(FILES package.xml DESTINATION share/${PROJECT_NAME})
+
+  endif()
+
+  # IF ROS2
+  if ($ENV{ROS_VERSION} EQUAL 2)
+    # Dependency to roscpp is also optional
+    cmake_policy(SET CMP0057 NEW)
+    # Default to C99
+    if(NOT CMAKE_C_STANDARD)
+      set(CMAKE_C_STANDARD 99)
```

Comment:
Why would we need this ? There is no C change in this PR. And actually… There is no C at all in this project, is there ?

Reply:
There is C++ as your generate the messages and services code in C++ (and compile it) and in Python.